Add the normal/pathogenic limits for each variant to the VCF

bjhall opened the pull request:
Adds the repeat number limits (max_normal and min_pathogenic) to each variant in the VCF.

We need this feature to be able to show these limits in Scout, to be able to quickly, for example, see if a pre-mutation is very close to normal or pathogenic.

Let me know if you have a better solution for this! I am also planning a pull request to Scout to show the limits in the STR list. I'll create an issue in the scout repo for that, so we can discuss it.

dnil commented on stranger/cli.py:
```diff
@@ -70,17 +86,20 @@ def cli(context, vcf, repeats_file, loglevel):
+        if repeat_data:
+            variant_info['info_dict']['STR_STATUS'] = repeat_data['repeat_strings']
+            variant_info['info_dict']['STR_NORMAL_MAX'] = str(repeat_data['lower'])
+            variant_info['info_dict']['STR_FULLMUT_MIN'] = str(repeat_data['upper'])
```
For our international contributors (and the poor guys who types in the values in the json definition file) could we keep this as PATHOLOGIC_MIN instead of FULLMUT_MIN!

dnil commented on stranger/utils.py:
```diff
@@ -135,8 +135,8 @@ def get_repeat_info(variant_info, repeat_info):
-    rep_lower = repeat_info[repeat_id]['normal_max']
-    rep_upper = repeat_info[repeat_id]['pathologic_min']
+    rep_lower = repeat_info[repeat_id].get('normal_max', 0)
+    rep_upper = repeat_info[repeat_id].get('pathologic_min', 0)
```
inf, -1, or some approximation? Will look strange for research vars w/o pathologic_min?

dnil commented on stranger/cli.py:
```diff
@@ -51,11 +51,27 @@ def cli(context, vcf, repeats_file, loglevel):
+            'desc': 'Max number of repeats allowed to call as normal'
+        },
+        {
+            'id': 'STR_FULLMUT_MIN', 'num': '1', 'type': 'Integer',
```
See below..

dnil commented on stranger/cli.py:
```diff
@@ -51,11 +51,27 @@ def cli(context, vcf, repeats_file, loglevel):
+            'desc': 'Repeat expansion status. Alternatives in [normal, pre_mutation, full_mutation]'
+        },
+        {
+            'id': 'STR_NORMAL_MAX', 'num': '1', 'type': 'Integer',
```
:+1:

dnil commented:
Mind reader there.. I was just saying we will do that any month now. 😄 
Could you update the CHANGELOG as well?

bjhall commented:
Thanks for your comments @dnil! I've changed FULLMUT to PATHOLOGIC, they were inconsistenly used throughout the code already, so I was unsure what to use. :)

I also changed the default (undefined) value to -1 instead of 0. That way Scout could just ignore the values. I don't know if "Inf" is allowed in an integer field in the VCF format?

dnil commented:
@bjhall have you tested this on your end? I was feeling trigger happy, but realise I probably should not press merge until someone has tested..

bjhall commented:
@dnil Absolutely, I'm an obsessive tester...
